src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1759,18 +1748,19 @@ macro_rules! read_word {
             $readed += size;
         }
         match size {
-            1 => buf[0] as i8 as i32 as u32,
-            2 => (((buf[0] as i8 as i32) << 8) + buf[1] as i32) as u32,
+            1 => i32::from(buf[0] as i8) as u32,
+            2 => ((i32::from(buf[0] as i8) << 8) + i32::from(buf[1])) as u32,
             3 => {
                 if buf[0] & 128 > 0 {
-                    (255 << 24) | ((buf[0] as u32) << 16) | ((buf[1] as u32) << 8) | (buf[2] as u32)
+                    (255 << 24) | (u32::from(buf[0]) << 16) | (u32::from(buf[1]) << 8)
+                        | u32::from(buf[2])
                 } else {
-                    ((buf[0] as u32) << 16) | ((buf[1] as u32) << 8) | (buf[2] as u32)
+                    (u32::from(buf[0]) << 16) | (u32::from(buf[1]) << 8) | u32::from(buf[2])
                 }
             }
             4 => {
-                (((buf[0] as i8 as i32) << 24) + ((buf[1] as i32) << 16) +
-                 ((buf[2] as i32) << 8) + (buf[3] as i32)) as u32
+                ((i32::from(u8::from(buf[0])) << 24) + (i32::from(buf[1]) << 16) +
```

Comment:
`u8`?

Reply:
Oh no, you're right.

Reply:
Hm, fixing that gives an error... See #2693.